### PR TITLE
Custom threshold metrics

### PR DESCRIPTION
**Pull request by topepo**
Replaces #37 

Adds custom metric sets for `threshold_perf()`.

### CODE REVIEW

**Comment by topepo** [comment]
Closes #25

File: R/threshold_perf.R [hunk 1]
```diff
@@ -8,12 +8,15 @@
 #' used to determine which level is the event of interest. For more details,
 #' see the Relevant level section of [yardstick::sens()].
 #'
-#' The currently calculated metrics are:
+#' The defualt calculated metrics are:
```
Comment (edgararuiz) [inline R/threshold_perf.R:11]: Typo, "default"

File: R/threshold_perf.R [hunk 2]
```diff
@@ -31,6 +34,13 @@
 #' of values between 0.5 and 1.0 are used. **Note**: if this
 #' argument is used, it must be named.
 #'
+#' @param metrics Either `NULL` or a [yardstick::metric_set()] with a list of
+#' performance metrics to calculate. The metrics should all be oriented towards
+#' hard class predictions (e.g. [yardstick::sensitivity()],
+#' [yardstick::accuracy()], [yardstick::recall()], etc.) and not
+#' class probabilities. A set of default metrics is used when `NULL` (see
+#' Details below).
+#'
```
Comment (edgararuiz) [inline R/threshold_perf.R:43]: Maybe add an example? 

```
resampled_threshold_perf <- mock_resamples %>%
  group_by(resample) %>%
  threshold_perf(Class, .pred_good, thresholds, metrics = metric_set(f_meas))
```

File: R/threshold_perf.R [hunk 3]
```diff
@@ -166,30 +184,31 @@ threshold_perf.data.frame <- function(.data,
...
+    # and add it on
+    sens_vec <- .data_metrics %>%
+      dplyr::filter(.metric == "sens") %>%
+      dplyr::pull(.estimate)
+
+    dist <- .data_metrics %>%
+      dplyr::filter(.metric == "spec") %>%
+      dplyr::mutate(
+        .metric = "distance",
+        # .estimate is spec currently. this recodes as distance
```
Comment (edgararuiz) [inline R/threshold_perf.R:206]: Sorry, I don't understand this comment

**Review by edgararuiz — APPROVED** [review]
Just a few small things

File: R/threshold_perf.R (hunk 3 above)
Comment (topepo) [reply R/threshold_perf.R:206]: We duplicated the specificity results (the `"spec"`) in the comment.

**Pull request merged**